### PR TITLE
The house edge now has a new secondary attack.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

https://github.com/tgstation/tgstation/assets/41715314/58ea5c95-fecd-42aa-b070-f8bd80cce5e4

The house edge weapon, a weapon made from the v8 engine from the black market, now has a slightly less awful secondary attack. For recap on how it already works, the house edge, while wielded, gives itself a charge for every successful melee hit that it lands. Un-wielding or dropping the house edge will reset it's charges. Each charge is visibly seen on the icon_state as a plume of fire coming out of the exhaust.

The old system had the house edge effectively launch the wielder as a dangerous tackle at it's target. This had a higher likelihood of just killing the wielder than it did to really provide any real advantage in a fight, against one or multiple targets, either way. At best, it stunned both targets, and dropped the house edge to the ground.

The new special attack instead launches the weapon as a throw from the wielder's hand, and then after 0.5-1.5 seconds, teleports the wielder to the sword, and puts it back in hand. It's range, as well as it's teleport delay is based on the number of fire charges, making it a particularly tricky weapon to use correctly. That said, it can allow for the wielder to effectively teleport out of a bad situation and buy some time for either an escape for an equipment swap as necessary. Using it this way also poses the potential risk that someone could catch the house edge, in which case the teleport will be cancelled. Plus it leaves behind a little fire teleport effect in the wielder's old location.

## Why It's Good For The Game

The original secondary effect was... godawful. I was made aware of it in the original PR, I've known it's basically a worse to use the secondary effect than to just use it as a discount fire axe. This helps to provide a useful buff to a rare weapon that provides some cool utility.

Having weapons that provide any kind of mobility is pretty rare, or blatantly comes with a higher bracket of power (ninja blade for example, mobility wands, etc). Having something like this that can live station-side, even if gated by both economy and rarity, would make this very cool and live in it's own unique niche.

## Changelog

:cl:
balance: The house edge, a rare blade made from an ancient V8 engine, now has a new special attack, activated by right clicking while wielded after landing one ore more successful hits.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
Maybe asking a lot but given that the throw essentially drops it _instantly_ on the destination tile it'd be nice to have some kind of trail and landing effect, maybe for the "following" teleport too.

**Comment by carlarctg** [comment]
I like the idea of a House Edge buff, but I don't like this implementation. The original is grounded in (cartoon) physics and fits a theme, it looks like an engine, you charge with it like an engine. This one.... you throw it? For some reason? And then you somehow teleport to it? Huh??
There are ways to buff the item that don't turn it into magic. It's built from a V8 engine, not magic unobtanium phlebutonium bluespace plasma or whatever.

**Comment by Jacquerel** [comment]
Yeah I guess that is also a reasonably good point, this just doesn't _look_ like a weapon which would teleport you
Maybe a charge which lets you phase through players without knocking either over and leaves some fire behind you would be more fitting 🤔

**Comment by necromanceranne** [comment]
I mean, not for nothing, but I think the charge is the much cooler ability if only it didn't punish you for collision. I think it should do the opposite of what it does currently. You get a cool big hit if you land a charge on someone, kinda like the grappling hooks launch. That's...what I thought the charge would do and was really surprised when it was decided otherwise. It just felt appropriate.

And, frankly, if you want to increase the possible design space for this, you could probably make it only lose charges after a delay from unequip/unwield, if only because I think it will at least alleviate how severely punishing the weapon can be at times, and lets you do more interesting things with unwielding the weapon on purpose.

...

If it did work like that...Boomerang component at max charges?

**Comment by YakumoChen** [comment]
Change the charge attack to make them move like a tram

**Comment by Admiral-Gigaton** [comment]
Skipping over other comments for now, what happens if the sword embeds on an enemy? (This could be used for funny/cool interactions, not unlike using recall spell when an item embeds)

And to address any extra things, such as comments about this not making sense - similar to another comment, you could add a trail from where the player is to where the item is, like the chef meat hook, and that trail being a... chain? Cable? (Likely this would change recipe for the item)

**Comment by Livrah** [comment]
Without any animation this looks weird and doesn't fit the theme if you ask me.

File: code/game/objects/items/v8_engine.dm [hunk 1]
```diff
@@ -91,15 +95,31 @@
...
+ * Teleport dash for the House Edge. Teleports the user to the sword and puts the sword back in the user's hand.
+ */
+/obj/item/house_edge/proc/teleport_dash(mob/living/user)
+	if(QDELETED(src) || user.incapacitated())
+		return FALSE
+	if(isnull(user) || ismob(loc))
+		reset_charges(on_dash = TRUE)
+		return FALSE
+	new /obj/effect/temp_visual/fire_fizzle(get_turf(user))
+	user.forceMove(drop_location())
```
Comment (MrMelbert) [inline code/game/objects/items/v8_engine.dm:118]: Also you're using forcemove rather than doteleport which means this can bypass noteleport stuff

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Should def. make it a "charge" instead of a teleport

**Comment by YuliceTheDruid** [comment]
The sword's a DMC reference, so a fiery charge would be more fitting thematically, yeah. Surely there'd be a way to give them some sort of unstoppable or unstunnable status while they're doing it, along with the fire trail

**Comment by Loafin34** [comment]
a cool idea for the charge is you wind it up after getting full charge, and get rush gland ability but can't stop moving until you hit something. once somethin is hit it takes 50 damage or you charge into a wall and take 10 damage and the sword is stuck in the wall, requiring you to grab at it to pull it out

**Comment by ArcaneMusic** [comment]
I'll come back to this some other time I'm currently overloaded with RL work to give this the creative energy it needs for this 5 seconds

**Pull request closed without merging**